tutorial/tutorial.md

```diff
@@ -0,0 +1,120 @@
+# Cloud Ops Agent Ansible Tutorial
+This tutorial will provide you with a preconfigured environment for deploying the new unified [Cloud Ops Agent](https://github.com/GoogleCloudPlatform/google-cloud-ops-agents-ansible). Inside the custom cloudshell environment, Ansible is already installed with the necessary inventory and configuration files to get started. To learn more about Ansible visit [docs.ansible.com](https://docs.ansible.com)
+
+## Prerequisites
+
+**NOTE:** By default, this tutorial applies the Cloud Ops Agent to all hosts within a project which may incur additional costs. See [Define Your Playbook](#define-your-playbook) below to change that behavior.
+
+#### In order to use Ansible and other configuration automation tools to install the Cloud Ops Agent you must:
+1. Have a service account available for Ansible to query the inventory
+2. Be able to ssh to the systems and have sufficient escalation privileges to install packages
+
+If these items are not the case for your project, consider creating a new test project following one of the [GCE Tutorials](https://cloud.google.com/compute/docs/tutorials) and see [SSH connections to Linux VMs](https://cloud.google.com/compute/docs/instances/ssh) for additional details.
+
+### Adding ssh keys
+If you have an SSH key that works for the GCE instances in this project, you can upload it inside this ephemeral container via Cloud Editor by going to *File > Upload File*
+
+## Service Account Setup 
+Confirm you are logged in:
+```bash
+ gcloud auth list
+ ```
+If not, login to your gcp account within cloudshell, this ensures you're able to run the necessary commands
+```bash
+gcloud auth login
+```
+Continue by setting your project ID to the project you'd like to work within. This project should have existing VMs available to run the Ansible playbook against:
+```bash
+gcloud config set project PROJECT_ID
+```
+Next you'll need to create a service account and credentials file. These resources allow configuration management tools to query and return the inventory from GCP dynamically. By using the `roles/compute.viewer` role, this credential can only view and compute resources and cannot modify any parts of the GCE system.
+1. Create the service account.
+```bash
+gcloud iam service-accounts create SERVICE_ACCOUNT_ID --description="DESCRIPTION" --display-name="DISPLAY_NAME"
+```
+2. Create the iam policy binding and attach the necessary role to the account:
+```bash 
+gcloud projects add-iam-policy-binding PROJECT_ID --member="serviceAccount:SERVICE_ACCOUNT_ID@PROJECT_ID.iam.gserviceaccount.com" --role="roles/compute.viewer"
+```
+3. Create the key-file associated with the service account:
+```bash
+gcloud iam service-accounts keys create key-file --iam-account=sa-name@project-id.iam.gserviceaccount.com
```

Comment:
i.e. this line could become 
`gcloud iam service-accounts keys create key-file --iam-account=$SERVICE_ACCOUNT_ID@$PROJECT_ID.iam.gserviceaccount.com`